plugins/modules/foreman_host.py

```diff
@@ -74,74 +116,48 @@
 
 RETURN = ''' # '''
 
-try:
-    from nailgun.entities import (
-        Host,
-        HostGroup,
-    )
-
-    from ansible.module_utils.ansible_nailgun_cement import (
-        find_host,
-        find_hostgroup,
-        find_location,
-        find_organization,
-        ForemanEntityAnsibleModule,
-        naildown_entity_state,
-        sanitize_entity_dict,
-    )
-except ImportError:
-    pass
-
-
-# This is the only true source for names (and conversions thereof)
-name_map = {
-    'name': 'name',
-    'enabled': 'enabled',
-    'hostgroup': 'hostgroup',
-    'location': 'location',
-    'organization': 'organization',
-}
+from ansible.module_utils.foreman_helper import ForemanEntityAnsibleModule
 
 
 def main():
     module = ForemanEntityAnsibleModule(
-        argument_spec=dict(
+        entity_spec=dict(
             name=dict(required=True),
-            hostgroup=dict(),
-            location=dict(),
-            organization=dict(),
+            hostgroup=dict(type='entity', flat_name='hostgroup_id'),
+            location=dict(type='entity', flat_name='location_id'),
+            organization=dict(type='entity', flat_name='organization_id'),
             enabled=dict(default='true', type='bool'),
```

Comment:
while not changed by you, what do you think about dropping the default from `enabled` too?

Reply:
agree